angular/projects/admin-nrpti/src/app/mines/mines-records-list/mines-records-list.component.ts

```diff
@@ -0,0 +1,566 @@
+    }
+    this.submit();
+  }
+
+  /**
+   * Page number changed (pagination).
+   *
+   * @param {*} pageNumber
+   * @memberof MinesRecordsListComponent
+   */
+  onPageNumUpdate(pageNumber) {
+    this.tableData.currentPage = pageNumber;
+    this.submit();
+  }
+
+  /**
+   * Page size picker option selected handler.
+   *
+   * @param {IPageSizePickerOption} pageSize
+   * @memberof MinesRecordsListComponent
+   */
+  onPageSizeUpdate(pageSize: IPageSizePickerOption) {
+    this.tableData.pageSize = pageSize.value;
+    this.submit();
+  }
+
+  /**
+   * Table multi-select handler.
+   *
+   * @param {*} rowData data for the row that was selected or unselected
+   * @param {boolean} checked whether or not the row was selected or unselected
+   * @memberof MinesRecordsListComponent
+   */
+  onRowCheckboxUpdate(rowData: any, checked: boolean) {
+    // TODO real multi-select functionality (adding to collections, etc)
+  }
+
+  /**
+   * Update mine table with latest values (whatever is set in this.tableData).
+   *
+   * @memberof MinesRecordsListComponent
+   */
+  submit() {
+    // These are params that should be handled by tableData
+    delete this.queryParams.sortBy;
+    delete this.queryParams.currentPage;
+    delete this.queryParams.pageNumber;
+    delete this.queryParams.pageSize;
+
+    this.loadingScreenService.setLoadingState(true, 'body');
+
+    this.router.navigate([{ ...this.queryParams, ...this.tableTemplateUtils.getNavParamsObj(this.tableData) }], {
+      relativeTo: this.route
+    });
+  }
+
+  toggleAdvancedFilters(): void {
+    this.showAdvancedFilters = !this.showAdvancedFilters;
+  }
+
+  changeSubset(filterText): void {
+    switch (filterText) {
+      case 'All':
+        this.selectedSubset = 'All';
+        delete this.queryParams['subset'];
+        break;
+      case 'Description & Summary':
+        this.selectedSubset = 'Description & Summary';
+        this.queryParams['subset'] = ['description'];
+        break;
+      case 'Issued To':
+        this.selectedSubset = 'Issued To';
+        this.queryParams['subset'] = ['issuedTo'];
+        break;
+      case 'Location':
+        this.selectedSubset = 'Location';
+        this.queryParams['subset'] = ['location'];
+        break;
+      case 'Record Name':
+        this.selectedSubset = 'Record Name';
+        this.queryParams['subset'] = ['recordName'];
+        break;
+      default:
+        break;
+    }
+    if (this.keywordSearchWords) {
+      this.queryParams.keywords = this.keywordSearchWords;
+      // always change sortBy to '-score' if keyword search is directly triggered by user
+      this.tableData.sortBy = '-score';
+      this.submit();
+    }
+  }
+
+  /**
+   * Set subset filters.
+   *
+   * @memberof MinesRecordsListComponent
+   */
+  setSubset() {
+    if (!this.queryParams.subset) {
+      this.selectedSubset = 'All';
+    } else if (this.queryParams.subset.includes('issuedTo')) {
+      this.selectedSubset = 'Issued To';
```

Comment:
Seeing these terms used a few times. Might move to a constants file?

Reply:
yeah can do